packages/synapse-react-client/src/components/CitationPopover/CitationPopover.test.tsx

```diff
@@ -0,0 +1,166 @@
+import { act, fireEvent, render, screen, waitFor } from '@testing-library/react'
+import { UseQueryResult, useQuery } from '@tanstack/react-query'
+import CitationPopover from './CitationPopover'
+
+jest.mock('@tanstack/react-query', () => ({
+  useQuery: jest.fn(),
+}))
+
+const mockUseQuery = useQuery as jest.MockedFunction<typeof useQuery>
+
+global.fetch = jest.fn(() =>
+  Promise.resolve({
+    ok: true,
+    status: 200,
+    statusText: 'OK',
+    text: () =>
+      Promise.resolve(
+        '@misc{test2025,\n  title = {Some BibTeX Entry},\n  year = {2025}\n}',
+      ),
+  }),
+) as jest.Mock
+
+const openPopover = () => {
+  const button = screen.getByText('Cite As')
+  fireEvent.click(button)
```

Comment:
Please use `click` from user event and avoid using `fireEvent.click` unless absolutely necessary (across this whole test file)

[More info](https://testing-library.com/docs/user-event/intro/#differences-from-fireevent)